modules/xmpp/RtxModifier.spec.js

```diff
@@ -0,0 +1,263 @@
+/*eslint-disable max-len*/
+/*jshint maxlen:false*/
+var RtxModifier = require("./RtxModifier.js");
+var SampleSdpStrings = require("./SampleSdpStrings.js");
+var transform = require('sdp-transform');
+var SDPUtil = require("./SDPUtil.js");
+
+var numVideoSsrcs = function (parsedSdp) {
+  let videoMLine = parsedSdp.media.find(m => m.type === "video");
+  return videoMLine.ssrcs
+    .map(ssrcInfo => ssrcInfo.id)
+    .filter((ssrc, index, array) => array.indexOf(ssrc) === index)
+    .length;
+};
+
+var getPrimaryVideoSsrc = function (parsedSdp) {
+  return getPrimaryVideoSsrcs(parsedSdp)[0];
+};
+
+// Only handles parsing 2 scenarios right now:
+// 1) Single video ssrc
+// 2) Multiple video ssrcs in a single simulcast group
+var getPrimaryVideoSsrcs = function (parsedSdp) {
+  let videoMLine = parsedSdp.media.find(m => m.type === "video");
+  if (numVideoSsrcs(parsedSdp) === 1) {
+    return [videoMLine.ssrcs[0].id];
+  } else {
+    let simGroups = getVideoGroups(parsedSdp, "SIM");
+    if (simGroups.length > 1) {
+      return;
+    }
+    let simGroup = simGroups[0];
+    return simGroup.ssrcs.split(" ")
+      .map(ssrcStr => parseInt(ssrcStr));
+  }
+};
+
+var getVideoGroups = function (parsedSdp, groupSemantics) {
+  let videoMLine = parsedSdp.media.find(m => m.type === "video");
+  videoMLine.ssrcGroups = videoMLine.ssrcGroups || [];
+  return videoMLine.ssrcGroups
+    .filter(g => g.semantics === groupSemantics);
+};
+
+describe ("RtxModifier", function() {
```

Comment:
unfortunately (or fortunately, maybe?) jasmine passes some context in a passed 'this', so some of the functions (`beforeEach` and `it`) are best using the old-style functions.  so our option is to use arrow syntax in some places but not others, or just be consistent with old-style functions everywhere.  what do you think?